test/image/mocks/z-new_tickmode_sync.json

```diff
@@ -0,0 +1,80 @@
+{
+   "data": [
+      {
+         "name": "Apples",
+         "type": "bar",
+         "x": [
+            "Jan",
+            "Feb",
+            "Mar",
+            "Apr",
+            "May"
+         ],
+         "y": [
+            232,
+            2506,
+            470,
+            1864,
+            -190
+         ]
+      },
+      {
+         "name": "Oranges",
+         "type": "scatter",
+         "x": [
+            "Jan",
+            "Feb",
+            "Mar",
+            "Apr",
+            "May"
+         ],
+         "y": [
+            -0.8,
+            0.09,
+            0.01,
+            0.13,
+            0.42
+         ],
+         "yaxis": "y2"
+      }
+   ],
+   "layout": {
+      "margin": {
+         "t": 40,
+         "r": 70,
+         "b": 40,
+         "l": 70
+      },
+      "width": 700,
+      "legend": {
+         "orientation": "h",
+         "x": 0.6,
+         "y": 1.1
+      },
+      "yaxis": {
+         "title": {
+            "text": "Apples"
+         },
+         "side": "left",
+         "range": [
+            0,
+            2506
+         ],
+         "minor": { "showgrid": true }
+      },
+      "yaxis2": {
+         "title": {
+            "text": "Oranges"
+         },
+         "side": "right",
+         "range": [
+            0,
+            0.42
+         ],
+         "overlaying": "y",
+         "tickmode": "sync",
+         "minor": { "showgrid": true },
```

Comment:
If I add  
```json
         "overlaying": "y2",
         "tickmode": "sync",
```
to `yaxis` instead, right now the result is not correct.

We should make sure the base axes are drawn before synced axes e.g. by sorting `args` in `draw.axes`.

Reply:
> > > Looking great.
> > > Now in order to fix the case of moving the base axis (which make other axes out of sync), you may change `args` of `axes.draw` function something like:
> > > ```js
> > > axes.draw = function(gd, arg, opts) {
> > >   if(args && args.length === 1) {
> > >     // add the ids of other axes if they are synced to this axis
> > >     args = args.concat(aboveList);
> > >   }
> > >   
> > >   ...
> > > };
> > > ```
> > 
> > 
> > @archmoj That's a really nice tip! I wouldn't do it without you! Done now :)
> 
> Getting closer. But I still think we should move the logic to the start of `axes.draw`. Right now the other axes do not update until mouse which could be fixed by that adjustment.

@archmoj Sorry but I tried to move it up in the callstack and no luck. I'm not sure what I can do in order to have what you want.

Reply:
Now that this case handled, let's add another subplots in the mock to ensure different orders are tested.
Something like this mock: https://github.com/plotly/plotly.js/blob/8f01627f17787728410562a70401a1d04345a53f/test/image/mocks/bar_display_height_zero.json

Let's test `sync` option for both x and y as well as the base axis comes first or last.
Thanks!

Reply:
Looks good to me! Done already